src/io/config.cpp

```diff
@@ -64,6 +64,20 @@ void GetBoostingType(const std::unordered_map<std::string, std::string>& params,
   }
 }
 
+void GetDataSampleStrategy(const std::unordered_map<std::string, std::string>& params, std::string* strategy) {
+  std::string value;
+  if (Config::GetString(params, "data_sample_strategy", &value)) {
+    std::transform(value.begin(), value.end(), value.begin(), Common::tolower);
+    if (value == std::string("goss")) {
+      *strategy = "goss";
+    } else if (value == std::string("bagging")) {
+      *strategy = "bagging";
```

Comment:
>strategy [](http://example.com/codeflow?start=7&length=8)

I guess you should check if the input parameter is a null pointer or not. Acutaully, why not use a reference instead of a pointer?

Reply:
We have several similar functions like `GetSampleStrategy` in our code base. Currently we don't check for `nullptr` because these code are not exposed to the users. But I think we can consider adding such checks all together in the future. Thanks!

Reply:
> we can consider adding such checks all together in the future

If there is work to improve this for the future and you believe that work should be done, can you please document it in a separate issue? So we don't have to remember such "we'll come back to it later" things from PR conversations.